scanpy/plotting/_tools/scatterplots.py

```diff
@@ -206,7 +206,7 @@ def embedding(
             and len(size) == adata.shape[0]
         ):
             size = np.array(size, dtype=float)
-    elif img_key is not None:
+    elif basis is "spatial":
```

Comment:
Could this behavior be determined by whether `sc.pl.spatial` was called, and not the key in `obsm`?

That is, if someone were to call `sc.pl.embedding(adata, basis="spatial")` I think that should use a scatter plot, but if someone called `sc.pl.spatial` that should plot the circles.